overlord/hookstate/ctlcmd/get_attr.go

```diff
@@ -0,0 +1,102 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+type getAttrCommand struct {
+	baseCommand
+
+	Positional struct {
+		Attributes []string `positional-arg-name:"<keys>" description:"option keys" required:"yes"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortGetAttrHelp = i18n.G("Prints values of interface attributes")
+var longGetAttrHelp = i18n.G(`
+The get command prints values of interface attributes for current connection.
+
+    $ snapctl get-attr serialpath
+    /dev/ttyS0
+
+If multiple attribute names are provided, a document is returned:
```

Comment:
The rationale is that returning a json document forces usage of a json parser, which is not the most straightforward experience when one is using shell command integration. That said, if you ask for multiple values at once, then you're already off into special behavior anyway.